### PR TITLE
Read expansion data prior to reading card data

### PR DESCRIPTION
**Pull request by flagrama**
When using the pio v2 data, the expansions are now in their own JSON file.
These changes read that data before reading card data so that the relevant
expansion data can be used to generate files. This should reduce the number of
manual changes that need to be made to YAML and groovy files

Also fixes the issue where using multiple pios would throw everything in the 
same files because of a naming issue due to not knowing the expansion name.

One concern to keep in mind, the Promo sets will be named *BlackStarPromos it seems, so care should be taken when generating new SWSH promo YAML/groovy/scans.

### CODE REVIEW

File: tools/src/main/java/tcgone/carddb/tools/Application.java [hunk 1]
```diff
@@ -67,6 +65,30 @@ public void run(ApplicationArguments args) throws Exception {
...
+        else {
+          expansionFileBuilder.append(part).append(File.separator);
+        }
+      }
+
+      for (String filename : pios) {
+        expansionIds.add(Paths.get(filename).getFileName().toString().split("\\.")[0]);
+      }
+
+      if (new File(expansionFileBuilder.toString()).isFile()) {
```
Comment (flagrama) [inline tools/src/main/java/tcgone/carddb/tools/Application.java:87]: Blah, this little trick doesn't work after moving some stuff around. So Kriby data is broken atm.

Comment (axpendix) [reply tools/src/main/java/tcgone/carddb/tools/Application.java:87]: What is the error?

**Review by axpendix — APPROVED** [review]

File: tools/src/main/java/tcgone/carddb/tools/PioReader.java [hunk 2]
```diff
@@ -100,9 +102,32 @@
...
+        for (PioSet ps : list) {
+          if (!Objects.equals(ps.id, expansionId)) continue;
+          expansion.name = ps.name;
+          expansion.id = "FILL_THIS";
+          expansion.abbr = ps.ptcgoCode;
+          expansion.enumId = ps.name.replace("–", "-").replace("’", "'").toUpperCase(Locale.ENGLISH)
+            .replaceAll("[ \\p{Punct}]", "_").replaceAll("_+", "_").replace("É", "E");
+          expansion.pioId = ps.id;
+        }
+        log.warn("PLEASE FILL IN id FIELD in {}", expansion.name);
```
Comment (axpendix) [inline tools/src/main/java/tcgone/carddb/tools/PioReader.java:123]: Is this mandatory or suggested? For mandatory stuff I recommend throwing an exception.

**Comment by axpendix** [comment]
Hi @flagrama do you think this is good to be merged?

**Comment by flagrama** [comment]
I believe this was in a working state? It's been a while though so I'm not 100%. It's probably fine.

**Review by axpendix — APPROVED** [review]

**Pull request merged**